go-controller/pkg/factory/handler.go

```diff
@@ -585,28 +560,27 @@ func newQueuedInformer(oType reflect.Type, sharedInformer cache.SharedIndexInfor
 	i.queueMap.start()
 
 	i.initialAddFunc = func(h *Handler, items []interface{}) {
-		// Make a handler-specific channel array across which the
-		// initial add events will be distributed. When a new handler
-		// is added, only that handler should receive events for all
-		// existing objects.
-		addsWg := &sync.WaitGroup{}
-		addsMap := newQueueMap(numEventQueues, addsWg, stopChan)
-		addsMap.start()
-
 		// Distribute the existing items into the handler-specific
 		// channel array.
 		for _, obj := range items {
-			addsMap.enqueueEvent(nil, obj, i.oType, false, func(e *event) {
+			objMeta, err := getObjectMeta(i.oType, obj)
+			if err != nil {
+				panic(err)
+			}
+			namespacedName := ktypes.NamespacedName{Namespace: objMeta.GetNamespace(), Name: objMeta.GetName()}
+			// we enqueue events on initial add directly, since on initial add we are holding informer lock
+			i.queueMap.enqueueEvent(nil, obj, i.oType, false, func(e *event) {
```

Comment:
We will see how much we relied on the order in which we process events on startup (i.e. nodes before pods)